packages/gatsby/cache-dir/navigation.js

```diff
@@ -97,10 +97,16 @@ const navigate = (to, options = {}) => {
   })
 }
 
-function shouldUpdateScroll(prevRouterProps, { location: { pathname } }) {
+function shouldUpdateScroll(prevRouterProps, routerProps) {
+  const {
+    location: { pathname, hash },
+  } = routerProps
   const results = apiRunner(`shouldUpdateScroll`, {
     prevRouterProps,
+    // `pathname` for backwards compatibility
     pathname,
+    routerProps,
+    savedScrollPositions: this._stateStorage,
```

Comment:
The naming here doesn't match the actual behavior. You don't get positions, you only get the storage.

Reply:
Yes, will update the docs with an example. Maybe it's enough to expose `getSavedScrollPosition: this._stateStorage.read`?

Reply:
That'd seem to be better, yeah.